File: C++/Find the N-th character.cpp
class Solution{
  public:
    char nthCharacter(string s, int r, int n) {
         
         while(r--)
         {
             string temp = "";
             for(int i=0;i<s.length();i++)
             {
                 if(s[i]=='1') temp+="10";
                 else temp+="01";
                 
                 if(temp.length()*pow(2,r)>=n+1) break;
             }
             s = temp ; 
         }
         return s[n];
    }
};

File: JAVA/Find the N-th character.java
class Solution
{
    public char nthCharacter(String s, int r, int n)
    {
       String answer = s;
int i=0;
while(i<r){
    String temp = "";
    for(int j=0;j<s.length();j++){
        if(answer.charAt(j)=='1'){
            temp += "10";
        }else if(answer.charAt(j) == '0'){
            temp += "01";
        }
    }
    answer = temp;
    i++;
}
return answer.charAt(n);
    }
}
